packages/block/src/block.ts

```diff
@@ -569,6 +569,10 @@ export class Block {
     let hf = ''
     try {
       hf = this._common.hardfork()
+      if (hf === Hardfork.Merge) {
+        // Add info of execHardFork if relevant
+        hf = `${hf}(${this._common.execHardfork()})`
```

Comment:
This feels dangerous to me since the merge hardfork has plenty of functional differences for block headers and I can imagine lots of places where we want to report that the block header error stems from the hardfork being `Merge` and not `GrayGlacier` or whatever previous fork.

Reply:
yes it still retains the merge hf now it comes as: `Merge (GrayGlacier)` so the user has more context